10-reading-data-into-R.Rmd

```diff
@@ -0,0 +1,275 @@
+* `read_log()`: Web log files
+
+Here is an example of loading a CSV file into R:
+
+```{r}
+#| eval: false
+library(readr)
+
+anes_csv <-
+  read_csv("anes_timeseries_2020_csv_20220210.csv")
+```
+
+This saves the CSV into the `anes_csv` object.
+
+### Loading Excel files into R
+
+Excel, the spreadsheet software program from Microsoft, is a common file format. We can load Excel spreadsheets into R using the {readxl} package. The {readxl} package supports both the legacy `.xls` format and the modern `.xlsx` format. 
+
+Let's review the options by running `??readxl::read_excel`:
+
+```
+read_excel(
+  path,
+  sheet = NULL,
+  range = NULL,
+  col_names = TRUE,
+  col_types = NULL,
+  na = "",
+  trim_ws = TRUE,
+  skip = 0,
+  n_max = Inf,
+  guess_max = min(1000, n_max),
+  progress = readxl_progress(),
+  .name_repair = "unique"
+)
+```
+
+We load our file into R by running `read_excel()` and specifying the arguments as necessary:
+
+```{r}
+#| eval: false
+library(readxl)
+
+anes_excel <-
+  read_excel(path = "anes_timeseries_2020_csv_20220210.xlsx")
+```
+
+The file is saved as a tibble in `anes_excel`.
+
+### Loading Stata, SAS, and SPSS files into R
+
+The {haven} package from the tidyverse imports a variety of proprietary data formats: Stata `.dta` files, SAS `.sas7bdat` and `.sas7bcat` files, and SPSS `.sav` files.
+
+Let's se how to load Stata files first. We can import `.dat` file formats using `haven::read_dat()`. Explore the `haven::read_dat()` options by searching for the help page, `??haven::read_dta`.
+
+```r
+read_dta(
+  file,
+  encoding = NULL,
+  col_select = NULL,
+  skip = 0,
+  n_max = Inf,
+  .name_repair = "unique"
+)
+```
+
+We load our file into R by running `read_dat()` and specifying the arguments as necessary:
+
+```{r}
+#| eval: false
+library(haven)
+
+anes_dta <-
+  read_dta(file = "anes_timeseries_2020_stata_20220210.dta")
+```
+
+The {haven} package has equivalent functions for SAS and SPSS: `read_sas()` and `read_sav()` respectively. This is how we load the ANES 2020 SAS file:
+
+```{r}
+#| eval: false
+library(haven)
+
+anes_sas <-
+  read_sas(file = "anes_timeseries_2020_sas_20220210.sas7bdat")
+```
+
+And the ANES 2020 SPSS file:
+
+```{r}
+#| eval: false
+library(haven)
+
+anes_sav <-
+  read_sav(file = "anes_timeseries_2020_spss_20220210.sav")
+```
+
+Generally, Stata, SPSS, and SAS files contain labels for variables and values. Using {haven} allows us to retain important information when analyzing tibbles in R.
+
+### Importing labeled data
+
+Stata, SPSS, and SAS files often contain labeled variables and values. Labels provide descriptive information about categorical data. With the {haven} package, we can import labels from Stata, SPSS, and SAS into R without loss of fidelity using a special class of data called `haven_labelled`.
+
+Taking a look at the ANES SPSS file:
+
+```{r, R.options=list(max.print = 6)}
+#| eval: FALSE
+library(dplyr)
+
+anes_sav %>% 
+  select(1:6) %>% 
+  glimpse()
+```
```

Comment:
It's probably good to evaluate this and the following to show what output looks like